allennlp/models/simple_tagger.py

```diff
@@ -0,0 +1,137 @@
+            These can either be integer indexes or one hot arrays of
+            labels, so of shape (batch_size, sequence_length) or of
+            shape (batch_size, sequence_length, vocabulary_size).
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        logits : torch.FloatTensor
+            A tensor of shape (batch_size, sequence_length, tag_vocab_size)
+            representing unnormalised log probabilities of the tag classes.
+        loss: : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+
+        """
+        batch_size = sequence_tokens.size()[0]
+        # TODO(Mark): Change to use NlpApi.
+        embedded_text_input = self.embedding(sequence_tokens)
+        encoded_text, _ = self.stacked_encoders(embedded_text_input)
+
+        logits = self.tag_projection_layer(encoded_text)
+        reshaped_log_probs = logits.view(-1, self.num_classes)
+        class_probabilities = F.softmax(reshaped_log_probs).view([batch_size, -1, self.num_classes])
+
+        output_dict = {"logits": logits, "class_probabilities": class_probabilities}
+
+        if sequence_tags:
+            # NLL criterion takes integer labels, not one hot.
+            if sequence_tags.dim() == 3:
+                _, sequence_tags = sequence_tags.max(-1)
+            loss = self.sequence_loss(reshaped_log_probs, sequence_tags.view(-1))
+            output_dict["loss"] = loss
+
+        return output_dict
+
+    def tag(self, text_field: TextField) -> Dict[str, numpy.array]:
+        """
+        Perform inference on a TextField to produce predicted tags and class probabilities
+        over the possible tags.
+
+        Parameters
+        ----------
+        text_field : ``TextField``, required.
+            A ``TextField`` containing the text to be tagged.
+
+        Returns
+        -------
+        A Dict containing:
+
+        tags : List[str]
+            A list the length of the text input, containing the predicted (argmax) tag
+            from the model per token.
+        class_probabilities : numpy.Array
+            An array of shape (text_input_length, num_classes), where each row is a
+            distribution over classes for a given token in the sentence.
+        """
+        text_field.index(self.vocabulary)
+        padding_lengths = text_field.get_padding_lengths()
+        array_input = text_field.pad(padding_lengths)
+        # TODO(Mark): Generalise how the array is transformed into a variable after settling the data API.
+        # Add a batch dimension by unsqueezing, because pytorch
+        # doesn't support inputs without one.
+        array_input = torch.autograd.Variable(torch.LongTensor(array_input[0])).unsqueeze(0)
+        output_dict = self.forward(sequence_tokens=array_input)
+
+        # Remove batch dimension, as we only had one input.
+        predictions = output_dict["class_probabilities"].data.squeeze(0)
+        _, argmax = predictions.max(-1)
```

Comment:
Nevermind, just read the docs more closely, and you're right.  That's confusing.  It doesn't help that the annotated return type in the docs is wrong...